include/aie/Dialect/AIEX/IR/AIEX.td

```diff
@@ -592,7 +613,7 @@ def AIE_NpuDmaMemcpyNdOp: AIEX_Op<"npu.dma_memcpy_nd", [
   let hasVerifier = 1;
 }
 
-def AIE_NpuDmaWaitOp: AIEX_Op<"npu.dma_wait", []> {
+def AIE_NpuDmaWaitOp: AIEX_Op<"npu.dma_wait", [HasParent<"RuntimeSequenceOp">]> {
```

Comment:
I'd prefer we not add this restriction to all the npu ops. They should be able to exist outside of a sequence. For example to support incremental lowering into and out of aie/aiex dialect. In fact I think there's a case to be made for the npu ops to live in their own standalone dialect. i.e. `npu.*` not `aiex.npu.*`

Reply:
Not sure I fully follow. What does `npu.write32` mean outside of a runtime sequence, for example?

Reply:
> Not sure I fully follow. What does `npu.write32` mean outside of a runtime sequence, for example?

It means the same, just in a different container.

Reply:
The container `aiex.runtime_sequence` in my mind means "this runs on the LX6/7". And I understand `write32(addr, value)` to be specifically an LX6/7 instruction that only it can execute. (Other parts of the AIE may write to registers, but they'll have different address spaces.)

What other containers would this instruction make sense in? Probably not `aie.core` and things like that, since they can't write to arbitrary other AIEs registers?

Reply:
> What other containers would this instruction make sense in? Probably not `aie.core` and things like that, since they can't write to arbitrary other AIEs registers?

`my_dialect.foo`, `func.func`, ...